src/core/service.c

```diff
@@ -1641,6 +1641,17 @@ static int service_spawn_internal(
                 }
         }
 
+        Job *j = UNIT(s)->job;
+        if (j && j->activation_event_info) {
+                r = activation_event_info_format(j->activation_event_info, &our_env);
```

Comment:
maybe call this `format_env()` rather than just `format()`. After all this is formatting as env vars, and not for example formatting for serialization purposes.

Reply:
maybe even activation_event_info_append_env() actually, to emphasize that the strv specified is extended, and not a simple return value